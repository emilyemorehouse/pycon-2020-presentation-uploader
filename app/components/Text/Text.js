/* eslint-disable react/prop-types */

import React from 'react'
import { Text } from 'grommet'
import styled from 'styled-components'

import { doc } from './Text.doc'
import helpers from '../../utils/helpers'

/**
 *
 * Text
 *
 */

const StyledText = styled(Text).attrs(props => ({ ...props }))`
  font-family: ${props => props.secondary && 'Lato'};
`
function createText({
  alignSelf,
  a11yTitle,
  as,
  gridArea,
  margin,
  children,
  color,
  level,
  responsive,
  size,
  textAlign,
  tag,
  truncate,
  weight,
  wordBreak,
  secondary,
  ...rest
}) {
  return (
    <StyledText
      secondary={secondary}
      alignSelf={alignSelf}
      a11yTitle={a11yTitle}
      as={as}
      gridArea={gridArea}
      margin={margin}
      color={color}
      level={level}
      responsive={responsive}
      size={size}
      textAlign={textAlign}
      tag={tag}
      truncate={truncate}
      weight={weight}
      wordBreak={wordBreak}
      {...rest}
    >
      {children}
    </StyledText>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createText,
})
