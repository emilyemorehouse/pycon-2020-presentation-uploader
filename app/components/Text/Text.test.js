import React from 'react'
import { Grommet, grommet } from 'grommet'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Text from './Text'

const renderComponent = (props = {}) =>
  render(
    <Grommet theme={grommet}>
      <Text {...props}>TESTING</Text>
    </Grommet>,
  )

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})
