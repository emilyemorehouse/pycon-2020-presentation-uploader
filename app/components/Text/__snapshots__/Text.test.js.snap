// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
.c1 {
  font-size: 18px;
  line-height: 24px;
}

.c0 {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial,sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 18px;
  line-height: 24px;
  background: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 "
    >
      TESTING
    </span>
  </div>
</div>
`;
